### PR TITLE
fixed terrain lod error for lodEnable change

### PR DESCRIPTION
**Pull request by MSoft1115**
Re: #

### Changelog

* fixed terrain lod error for lodEnable change
* use stanard light map uniform for terrain

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/terrain/terrain.ts [hunk 1]
```diff
@@ -196,6 +196,10 @@ class TerrainRenderable extends ModelRenderer {
      * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
      */
     public _currentMaterialLayers = 0;
+    /**
+     * @deprecated since v3.5.0, this is an engine private interface that will be removed in the future.
```
Comment (SantyWang) [inline cocos/terrain/terrain.ts]: no need to add ```@deprecated```，use ```@engineInternal``` instead

File: editor/assets/effects/builtin-terrain.effect [hunk 2]
```diff
@@ -265,7 +262,7 @@ CCProgram terrain-fs %{
     s.emissive = vec3(0.0, 0.0, 0.0);
 
     #if CC_USE_LIGHTMAP && !USE_BATCHING && !CC_FORWARD_ADD
-      vec4 lightColor = texture(lightMap, luv.xy);
+      vec4 lightColor = texture(cc_lightingMap, luv.xy);
```
Comment (SantyWang) [inline editor/assets/effects/builtin-terrain.effect:265]: Are there compatibility issues when user upgrade from old version

Comment (MSoft1115) [reply editor/assets/effects/builtin-terrain.effect:265]: > Are there compatibility issues when user upgrade from old version

it's compatibility for old version

File: cocos/terrain/terrain.ts (hunk 1 above)
Comment (MSoft1115) [reply cocos/terrain/terrain.ts]: This is automatically added

File: cocos/terrain/terrain.ts [hunk 3]
```diff
@@ -578,13 +582,22 @@ export class TerrainBlock {
...
-                mtl.setProperty('lightMap', this.lightmap);
-                mtl.setProperty('lightMapUVParam', this.lightmapUVParam);
+            if (this._renderable._model && this.lightmap !== this._renderable._lightmap) {
+                this._renderable._lightmap = this.lightmap;
+                this._renderable._model?.updateLightingmap(this.lightmap, this.lightmapUVParam);
             }
         }
     }
 
+    public _buildLodInfo() {
```
Comment (SantyWang) [inline cocos/terrain/terrain.ts:595]: Can be private ?

Comment (MSoft1115) [reply cocos/terrain/terrain.ts:595]: > Can be private ?
Added @engineInternal tag

**Review by SantyWang — APPROVED** [review]

**Pull request merged**